src/lib/userStorage/FeedStorage.js

```diff
@@ -444,49 +446,48 @@ export class FeedStorage {
     return
   }
 
-  updateFeedEventCounterParty(feedEvent) {
-    const getCounterParty = async address => {
-      if (!address) {
-        return
-      }
-      let { fullName, smallAvatar } = await this.userStorage.getPublicProfile(address)
-
-      if (!(fullName || smallAvatar)) {
-        return
-      }
-
-      /** THIS CODE BLOCK MAY BE REMOVED AFTER SEPTEMBER 2021 */
-      /** =================================================== */
-      if (Config.ipfsLazyUpload && smallAvatar) {
-        // keep old base64 value if upload failed
-        smallAvatar = await updateFeedEventAvatar(smallAvatar).catch(() => smallAvatar)
-      }
-
-      /** =================================================== */
-
-      assign(feedEvent.data, {
-        counterPartyAddress: address,
-        counterPartyFullName: fullName,
-        counterPartySmallAvatar: smallAvatar,
-      })
-
-      await this.updateFeedEvent(feedEvent)
-    }
+  async updateFeedEventCounterParty(feedEvent) {
+    let addressField
 
     log.debug('updateFeedEventCounterParty:', feedEvent.data.receiptEvent, feedEvent.id, feedEvent.txType)
 
     switch (feedEvent.type) {
       case FeedItemType.EVENT_TYPE_SENDDIRECT:
       case FeedItemType.EVENT_TYPE_SEND:
-        getCounterParty(get(feedEvent, 'data.receiptEvent.to'), feedEvent)
+        addressField = 'to'
         break
       case FeedItemType.EVENT_TYPE_WITHDRAW:
       case FeedItemType.EVENT_TYPE_RECEIVE:
-        getCounterParty(get(feedEvent, 'data.receiptEvent.from'), feedEvent)
+        addressField = 'from'
         break
       default:
         break
     }
+
+    if (!addressField) {
+      return
+    }
+
+    const address = get(feedEvent, `data.receiptEvent.${addressField}`)
+    let { name, avatar } = await this.userStorage.getUserProfile(address)
+
+    /** THIS CODE BLOCK MAY BE REMOVED AFTER SEPTEMBER 2021 */
+    /** =================================================== */
+    if (Config.ipfsLazyUpload) {
+      // keep old base64 value if upload failed
+      avatar = await updateFeedEventAvatar(avatar).catch(() => avatar)
+    }
+
+    /** =================================================== */
+
+    assign(feedEvent.data, {
+      counterPartyAddress: address,
+      counterPartyFullName: name,
+      counterPartySmallAvatar: avatar,
+      counterPartyLastUpdate: new Date().getTime(),
```

Comment:
this shouldnt be saved in backend
a local cache by address should store that
you can make the table with address as key and add the cid hash as index

Reply:
@sirpy 
1. counterPartyLastUpdate  is stored in the Feed collection of the thread db 
i checked - on backend we're storing some different and encrypted data.
i could re check and make sure counterPartyLastUpdate won't be stored on backend. but this is not so critical from my point of view.

2. no need to create a separate table as we could add just one property for the existing one. This is more easiest and simple solution as you like.

Reply:
@sirpy or you want to load & cache this data by counterparty address ? e.g. if we have a few tx with the same counterparty - load its data once, cache and periodically reload (by ttl) until he/she will have name + avatar ?

Reply:
1. on threaddb it is saved unencrypted but all changes are synced to backend encrypted. so any change also incures a backend call and i dont want that.
2. I didnt say we need a new table ,we just need the assets table

yes exactly. counterparty details is always the same for all txs. no need to periodically reload. we try to periodically reload if counterparty data is empty.

Reply:
ok will look onto it